app/code/Magento/Sales/Controller/Adminhtml/Order/Creditmemo/View.php

```diff
@@ -75,9 +75,8 @@ public function execute()
             }
             return $resultPage;
         } else {
-            $resultForward = $this->resultForwardFactory->create();
-            $resultForward->forward('noroute');
-            return $resultForward;
+            $resultRedirect = $this->resultRedirectFactory->create();
+            return $resultRedirect->setPath('sales/creditmemo');
```

Comment:
And instead of using `} else {` block - surround `$creditmemo = $this->creditmemoLoader->load()` with `try { } catch (NoSuchEntityException $e) {` and handle the error with redirect and adding `$this->messageManager->addErrorMessage(__('This creditmemo no longer exists.'));` there.